compiler/daml-extension/BUILD.bazel

```diff
@@ -3,77 +3,65 @@
 
 package(default_visibility = ["//visibility:public"])
 
-load("@npm_bazel_typescript//:index.bzl", "ts_library")
-load("@build_bazel_rules_nodejs//:defs.bzl", "npm_package")
-load("//bazel_tools:pkg.bzl", "pkg_tar")
-
-ts_library(
-    name = "daml_extension_lib",
-    # TODO(MH): Unfortunately, the current packaging setup does not work with
-    # multiple source files. We need to figure out how to split this file up
-    # before it gets too big and package it properly.
-    srcs = ["src/extension.ts"],
-    node_modules = "@daml_extension_deps//:node_modules",
-    tsconfig = ":tsconfig.json",
-    deps = [
-        "@daml_extension_deps//@types",
-        "@daml_extension_deps//vscode",
-        "@daml_extension_deps//vscode-languageclient",
-    ],
+sh_test(
+    name = "valid-json",
+    srcs = ["ci-tests.sh"],
+    data = glob([
+        "*.json",
+        "syntaxes/*.json",
+    ]),
 )
 
-# With this rule we get access to extension.js, as
-# the ts_library only has the .d.ts file in the outputs.
-# Could possibly also use filegroup, which allows specifying
-# the output group (es5_source).
-npm_package(
-    name = "out",  # named out, so it goes to same place as before
-    deps = [
-        ":daml_extension_lib",
-    ],
+genrule(
+    name = "node_deps_cache",
```

Comment:
Is it actually worth caching this separately? How slow is it? The extension doesn’t change too often so the cases where the yarn lock file doesn’t change while the extension changes (which would be the only case where this helps) should be quite rare.

Reply:
Takes about 80s on my machine, whereas the rest (which required most of the iterating) runs in about 3s. It was definitely worth separating for writing these rules. I suppose once it all works we may go back and merge them, but I'm not sure what harm it makes to hacve this as a separate step?

Reply:
No harm, I was just thinking about reducing the complexity. But given those timings it seems quite reasonable to have this as a separate step. That said, I am not sure we gain much by splitting this into one rule that wraps `node_modules` in a tarball only to then extract it in the other rule so I would just inline the logic for this. The actual caching should come from copying `@daml_extension_deps//:node_modules` not from the fact that this is a separate rule.

Reply:
I totally agree with you, but my computer doesn't. Getting the files from Bazel takes 80s, getting the tarball from Bazel, extracting it, _and_ doing all the other stuff with yarn & vsce takes 5s.

Reply:
I’m confused, you are saying if you copy from `@daml_extension_deps//:node_modules` to `node_modules` before calling `yarn` instead of going via the indirection it gets slower?

Reply:
From my observations, it looks like what is slow is getting files from Bazel (specifically lots of small files). I suspect this has something to do with symlinks and may be faster on Linux.

So getting `node_modules` one file at a time, as currently done in the `node_deps_cache` rule, takes a _long_ time (we're talking ~80s on my machine for about 60MB across 3k files), whereas fetching one tarball from Bazel and extracting that "locally" (as currently done by the `vsix` rule) is about as fast as I'd expect, i.e. practically instant.

I don't really have a good explanation for why (except that hunch about symlinks), but there it is. I do not observe any speed-up attributable to caching if I inline the logic from `node_deps_cache` into `vsix`*. This is in fact what I started with; this is not a case of premature optimization, I had written the slow version first.

*: Other than caching the whole result of the genrule, obviously. But when I change the text of the genrule, even for steps that happen strictly after that point, I get the full 80s again if the fetching of files is inline.

Reply:
> I’m confused, you are saying if you copy from `@daml_extension_deps//:node_modules` to `node_modules` before calling `yarn` instead of going via the indirection it gets slower?

More specifically: the copying is what is slow, hence the creation of a separate rule that just does the copying, so that gets cached as a single tarball and on subsequent runs I only have to pay the cost of accessing a Bazel-provided file once (rather than 3k times).

Reply:
Thank you for the clarification! I definitely did not expect that copying a bunch of files takes 80s which is why I got a bit confused. I completely agree that a separate rule is the right way to handle this in that case!